lib/miq_automation_engine/engine/miq_ae_method.rb

```diff
@@ -104,13 +115,21 @@ def self.invoke_external(cmd, workspace, serialize_workspace = false)
         end
         rc  = status.exitstatus
         msg = "Method exited with rc=#{verbose_rc(rc)}"
+        method_pid = nil
+        threads = []
       rescue => err
         $miq_ae_logger.error("Method exec failed because (#{err.class}:#{err.message})")
         rc = MIQ_ABORT
         msg = "Method execution failed"
+      ensure
+        if method_pid
+          $miq_ae_logger.error("Terminating non responsive method with pid #{method_pid.inspect}")
+          Process.kill("TERM", method_pid) rescue nil
```

Comment:
yeah, what are we rescuing?  

From: http://ruby-doc.org/core-2.2.0/Process.html#method-c-kill
```
If signal is an integer but wrong for signal, Errno::EINVAL or RangeError will be raised. Otherwise unless signal is a String or a Symbol, and a known signal name, ArgumentError will be raised.

Also, Errno::ESRCH or RangeError for invalid pid, Errno::EPERM when failed because of no privilege, will be raised. In these cases, signals may have been sent to preceding processes.
```

I'd prefer something like this:

```ruby
begin
  Process.kill("TERM", method_pid) && Process.wait(method_pid)
rescue Errno::ESRCH, RangeError
  # Process doesn't exist
end
```